plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -491,17 +530,26 @@ private static String getGeneratedColumnExpressions(JsonNode node)
         return generationExpression == null ? null : generationExpression.asText();
     }
 
-    public static Map<String, String> getCheckConstraints(MetadataEntry metadataEntry)
+    public static Map<String, String> getCheckConstraints(MetadataEntry metadataEntry, ProtocolEntry protocolEntry)
     {
+        if (protocolEntry.supportsWriterFeature() && !protocolEntry.writerFeaturesContains("checkConstraints")) {
+            return ImmutableMap.of();
```

Comment:
Why we check for checkConstraints now and we didn't before?
If the previous logic was flawed, and this is a fix, it would be nice to have it as a separate commit.

Reply:
We didn't check because the previous max writer version 6 shouldn't have writer features. The empty writer features in versions <= 6 is ensured in `ProtocolEntry` class. I don't think the previous logic was flawed. 

Same for `invariants` & `identityColumns`.